[fritzboxtr064] add items for dsl/wan statistics (#5221)

tensberg opened the pull request:
This change adds items for DSL and WAN statistics to the fritzboxtr064 binding.

Supporting the new items required a somewhat large change to the ItemMap-related code because previously the code did not support reading the values of multiple items in a single SOAP service request. I basically splitted the existing ItemMap in a MultiItemMap which can extract multiple item values from a single SOAP response but does not (currently) support input parameters and writing values, and a SingleItemMap, which has all existing functionality but supports only a single item value per SOAP response. I also introduced the ItemConfiguration abstraction for the openhab item config to make the handling of item commands and parameters explicit

I asked gitbock if he agrees with me making changes to the binding. He is fine with it as he is no longer actively working on it.

tensberg commented on bundles/binding/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/AbstractItemMap.java:
```diff
@@ -0,0 +1,50 @@
+/***
+ * Abstract base implementation of an {@link ItemMap}.
+ *
+ * @author gitbock
```
I refactored the ItemMap into the AbstractItemMap, so most of the code is still from gitbock. Since I made so many changes I now changed the author to me. After all, if I broke it, I should take the blame...

tensberg commented:
Thanks for the review! I made the changes to the documentation and the log messages which you requested.

Reviewing the code which generates the log messages I noticed that the SoapValueParser-related code could use some rework to improve and unify the error handling. I added this a separate commit so you only have to review the latest changes, but intend to squash it into commit 2de7a5d afterwards because it obsoletes the classes Default- and SingleSoapValueParser which I introduced in that commit.

I also rebased the branch onto the latest master to fix the merge conflicts.

9037568 commented:
Looks like a couple of the files need an Eclipse formatting. Other than that, I think this is good to go.

tensberg commented:
I rebased the branch on the latest master and squashed some commits which belonged together. I also applied the Eclipse ESH formatter on all Java files which I touched, but it changed almost nothing.
@9037568: could you give an example of a code location which needs formatting?

9037568 commented:
These changes fixed the formatting. Thanks!
